crates/futures/src/lib.rs

```diff
@@ -101,12 +101,21 @@
 //! }
 //! ```
 
+#![feature(stdsimd)]
 #![deny(missing_docs)]
 
 #[cfg(feature = "futures_0_3")]
 /// Contains a Futures 0.3 implementation of this crate.
 pub mod futures_0_3;
 
+#[cfg(target_feature = "atomics")]
+/// Contains a thread-safe version of this crate, with Futures 0.1
+pub mod atomics;
+
+#[cfg(target_feature = "atomics")]
+/// Polyfill for `Atomics.waitAsync` function
+mod polyfill;
+
 use std::cell::{Cell, RefCell};
```

Comment:
Oh so actually the existing implementation in this file should be removed when `atomics` is enabled. With atomics the current implementation is unsound, but the `atomics` module contains the same interface anyway so the idea is that we'd just swap out for different implementations depending on `atomics` or not.